packages/core/src/test/awsService/ec2/activation.test.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import * as sinon from 'sinon'
+import { assertTelemetry } from '../../testUtil'
+import { Ec2InstanceNode } from '../../../awsService/ec2/explorer/ec2InstanceNode'
+import { Ec2ParentNode } from '../../../awsService/ec2/explorer/ec2ParentNode'
+import { Ec2Client } from '../../../shared/clients/ec2Client'
+import { Ec2Connecter } from '../../../awsService/ec2/model'
+import { PollingSet } from '../../../shared/utilities/pollingSet'
+
+describe('ec2 activation', function () {
+    let testNode: Ec2InstanceNode
+
+    before(function () {
+        const testRegion = 'test-region'
+        const testPartition = 'test-partition'
+        // Don't want to be polling here, that is tested in ../ec2ParentNode.test.ts
+        // disabled here for convenience (avoiding race conditions with timeout)
+        sinon.stub(PollingSet.prototype, 'start')
+        const testClient = new Ec2Client(testRegion)
+        const parentNode = new Ec2ParentNode(testRegion, testPartition, new Ec2Client(testRegion))
+        testNode = new Ec2InstanceNode(parentNode, testClient, testRegion, testPartition, {
+            InstanceId: 'testId',
+            LastSeenStatus: 'status',
+        })
+    })
+
+    after(function () {
+        sinon.restore()
+    })
+    it('terminal open', async function () {
+        const terminalStub = sinon.stub(Ec2Connecter.prototype, 'attemptToOpenEc2Terminal')
+        await vscode.commands.executeCommand('aws.ec2.openTerminal', testNode)
+
+        assertTelemetry('ec2_connectToInstance', { ec2ConnectionType: 'ssm' })
+        terminalStub.restore()
+    })
+
+    it('remote window open', async function () {
+        const remoteWindowStub = sinon.stub(Ec2Connecter.prototype, 'tryOpenRemoteConnection')
+        await vscode.commands.executeCommand('aws.ec2.openRemoteConnection', testNode)
+
+        assertTelemetry('ec2_connectToInstance', { ec2ConnectionType: 'remoteWorkspace' })
+        remoteWindowStub.restore()
+    })
```

Comment:
if these tests have stubbed the command so that it does nothing useful, then they could all be one `it('telemetry')` test.

Alternatively, if there are existing tests for these commands, can't those existing tests call `assertTelemetry` ?

Reply:
The high-level commands are simply wrappers around the logic that is tested in https://github.com/aws/aws-toolkit-vscode/blob/142761e268dd48687d0f696f6ded22819c74ffc7/packages/core/src/test/awsService/ec2/model.test.ts. The only difference at this level is the prompting logic (tested in https://github.com/aws/aws-toolkit-vscode/blob/142761e268dd48687d0f696f6ded22819c74ffc7/packages/core/src/test/awsService/ec2/prompter.test.ts) and telemetry (tested here). It doesn't seem to me to make sense to test any E2E aspects of the code beyond telemetry here.